internal/locate/region_request.go

```diff
@@ -391,36 +393,57 @@ type tryFollower struct {
 	stateBase
 	leaderIdx AccessIndex
 	lastIdx   AccessIndex
+	// fromOnNotLeader indicates whether the state is changed from onNotLeader.
+	fromOnNotLeader bool
 }
 
 func (state *tryFollower) next(bo *retry.Backoffer, selector *replicaSelector) (*RPCContext, error) {
 	var targetReplica *replica
+	hasDeadlineExceededErr := false
```

Comment:
Ditto, we may need a way to distinguish timeout error using `input timeout value` or the default timeout value, maybe abstract another error type for the configurable timeout error?